samples/snippets/cloud-client/src/test/java/privateca/SnippetsIT.java

```diff
@@ -133,32 +143,43 @@ public static void sleep(int seconds) throws InterruptedException {
   }
 
   // Create an asymmetric key pair to be used in certificate signing.
-  public static ByteString createPublicKey()
+  public static Map.Entry createPemEncodedKeyPair()
```

Comment:
This feels like a hacky use for Map.Entry since a) this isn't part of a map, b) this isn't a key-value pair, and most importantly c) it's not clear from the return type which part is the private key and which is the public key.

I would suggest simply returning the KeyPair from this method and pem-encoding it in the calling method above. This leaves a clear contract and makes it easy to follow which is the private vs. public key. The calling method seems to have to do some pem-encoding anyway so this would fit in nicely.

Reply:
I had an ambivalent emotion about exposing too much of BC framework in the setup, and hence had to take the less prettier way.
But I totally agree with your viewpoint.